blktap/blktap.cgroups

```diff
@@ -0,0 +1,7 @@
+{
+   "memory": {
+     "MaxPercentageRam": 25.0
```

Comment:
Also we might want to set `memory.swampiness` to some low value (e.g. `1`, I wouldn't set it to 0 because that'd invoke the OOM killer, swapping might still be preferable to OOM killing).

We probably need to find a good value through experimentation, or dynamically adjust based on memory pressure (not sure if that is possible). 25% seems quite low, after all if you give Dom0 extra RAM it might be because you want more available for read caching. An idle system seems to have ~85% free Dom0 RAM (including pagecache), OTOH one with lots of qemus might have less. 75% might be a better starting point.

One possible experiment could be:
* open a large file over NFS RO, and read from it in parallel from multiple processes (e.g. with appropriate `fio` flags in a loop and lots of processes in the hundreds, simulating VMs accessing a base disk)
* in parallel that do something that allocates/frees memory with periodic pauses, e.g. several VDI exports, and parallel VM writes (these would allocate tapdisk buffers, and free them again when idle)

I would expect there is MaxPercentageRam below 100 where it all works well, and above which you start seeing pauses/latency spikes. Then we can pick a value below that high threshold as default (and ideally add a xenrt test, but for now I'd be happy with just manual confirmation).

Reply:
Also it might be useful to be able to override this at runtime for easier experimentation (and also to solve customer issues without doing unsupported modifications to Dom0 fs). E.g. VBD polling thresholds are configurable through XAPI.
Maybe we could use a field in `sm_config` to configure this (and of course have a suitable default if that is not set)

Reply:
Hi Edvin,  using the new implementation, we just need to edit the vm-blktap.slice file and then do a "systemctl enable vm-blktap.slice" and a "systemctl restart vm-blktap.slice". Of course this would change the behaviour of all tapdisks.

If you want to configure individual tapdisks that would be way more complex.

Reply:
Once people think this approach is correct, I can start to run some experiments to determine the correct ram percentage.